src/app/dialog/dialog-service/dialog.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Observable } from 'rxjs/Rx';
+import { MdDialogRef, MdDialog, MdDialogConfig } from '@angular/material';
+import { Injectable } from '@angular/core';
+
+import { AlertDialogComponent, AlertDialogConfiguration } from './alert-dialog/alert-dialog.component';
+import { ConfirmDialogComponent, ConfirmDialogConfiguration } from './confirm-dialog/confirm-dialog.component';
+import { AskDialogConfiguration, AskDialogComponent } from './ask-dialog/ask-dialog.component';
+import { isUndefined } from 'util';
```

Comment:
let's just stick to conventional ways of checking for undefined. This one is Node module and is deprecated is the latest version